test/GroupByTest.cpp

```diff
@@ -1912,3 +1911,282 @@ TEST(GroupBy, Descriptor) {
+  std::vector<IdTable> idTables;
+  idTables.push_back(makeIdTableFromVector({{1}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{2}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{4}}, IntId));
+
+  auto subtree = makeExecutionTree<ValuesForTesting>(
+      qec_, std::move(idTables), vars("?x"), true, std::vector<ColumnIndex>{0});
+  Alias alias{SparqlExpressionPimpl{makeSum("?x"), "SUM(?x)"}, V{"?sum"}};
+  GroupBy groupBy{qec_, {}, {std::move(alias)}, std::move(subtree)};
+
+  expectReturningIdTables<1>(groupBy, {makeIdTableFromVector({{7}}, IntId)});
+}
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture,
+       verifyGroupsSpanningOverMultipleIdTablesAreAggregatedCorrectly) {
+  std::vector<IdTable> idTables;
+  idTables.push_back(makeIdTableFromVector({{1, 0}, {2, 1}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{4, 1}, {8, 1}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{16, 1}, {32, 2}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{64, 2}, {128, 2}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{256, 3}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{512, 3}}, IntId));
+
+  auto subtree = makeExecutionTree<ValuesForTesting>(
+      qec_, std::move(idTables), vars("?x", "?y"), true,
+      std::vector<ColumnIndex>{1});
+  Alias alias{SparqlExpressionPimpl{makeSum("?x"), "SUM(?x)"}, V{"?sum"}};
+  GroupBy groupBy{qec_, {V{"?y"}}, {std::move(alias)}, std::move(subtree)};
+
+  expectReturningIdTables<4>(groupBy,
+                             {makeIdTableFromVector({{0, 1}}, IntId),
+                              makeIdTableFromVector({{1, 30}}, IntId),
+                              makeIdTableFromVector({{2, 224}}, IntId),
+                              makeIdTableFromVector({{3, 768}}, IntId)});
+}
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture,
+       verifyGroupsAtTableBoundariesAreAggregatedCorrectly) {
+  std::vector<IdTable> idTables;
+  idTables.push_back(makeIdTableFromVector({{1, 0}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{2, 1}, {4, 1}}, IntId));
+  idTables.push_back(makeIdTableFromVector({{8, 2}, {16, 2}}, IntId));
+  auto subtree = makeExecutionTree<ValuesForTesting>(
+      qec_, std::move(idTables), vars("?x", "?y"), true,
+      std::vector<ColumnIndex>{1});
+  Alias alias{SparqlExpressionPimpl{makeSum("?x"), "SUM(?x)"}, V{"?sum"}};
+  GroupBy groupBy{qec_, {V{"?y"}}, {std::move(alias)}, std::move(subtree)};
+
+  expectReturningIdTables<3>(groupBy,
+                             {makeIdTableFromVector({{0, 1}}, IntId),
+                              makeIdTableFromVector({{1, 6}}, IntId),
+                              makeIdTableFromVector({{2, 24}}, IntId)});
+}
+
+// _____________________________________________________________________________
+TEST_P(GroupByLazyFixture, verifyImplicitGroupByWorksWithEmptyInput) {
```

Comment:
The naming of the tests is a little redundante.
Tests always check, test, or verify something, so that part is redundant. And `WorksWith`is also the default 
(negative tests should be named explicitly).
So this test case is maybe `TEST_P(Fixture, implicitGroupByWithEmptyInput)`
(Also check the remaining test cases.